yt_dlp/extractor/itv.py

```diff
@@ -105,28 +120,69 @@ def _real_extract(self, url):
                 href = ios_base_url + href
             ext = determine_ext(href)
             if ext == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
+                hls_formats, hls_subs = self._extract_m3u8_formats_and_subtitles(
                     href, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id='hls', fatal=False))
+                    m3u8_id='hls', fatal=False)
+                formats.extend(hls_formats)
             else:
                 formats.append({
                     'url': href,
                 })
         self._sort_formats(formats)
 
         subtitles = {}
-        subs = video_data.get('Subtitles') or []
-        for sub in subs:
-            if not isinstance(sub, dict):
-                continue
-            href = url_or_none(sub.get('Href'))
-            if not href:
-                continue
-            subtitles.setdefault('en', []).append({
-                'url': href,
-                'ext': determine_ext(href, 'vtt'),
-            })
-
+        platform_tag_subs = None
+        featureset_subs = None
+        for platform_tag, featuresets in variants.items():
+            for featureset in featuresets:
+                if (isinstance(featureset, list) and len(featureset) >= 3
+                        and featureset[2] == 'outband-webvtt'):
+                    platform_tag_subs = platform_tag
+                    featureset_subs = featureset
+                    break
+        if not platform_tag_subs or not featureset_subs:
+            self.report_warning('No subtitles available', video_id)
```

Comment:
There is no such warning for other extractors. This isnt needed. If the user tries to download subs, the core code will issue a warning

Reply:
Also, since the subtitle extraction code requires additional requests, it's probably better to move this into `_get_subtitles` and do `'subtitles': self.extract_subtitles(...)`